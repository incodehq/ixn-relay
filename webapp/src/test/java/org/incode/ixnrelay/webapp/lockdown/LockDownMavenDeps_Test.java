package org.incode.ixnrelay.webapp.lockdown;

import java.io.IOException;
import java.net.URL;
import java.util.Arrays;

import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Before;
import org.junit.Test;

import com.google.common.base.Charsets;
import com.google.common.io.Resources;

import static org.hamcrest.CoreMatchers.*;
import static org.junit.Assume.assumeThat;

public class LockDownMavenDeps_Test {

    @Before
    public void setUp() throws Exception {
        assumeThat(System.getProperty("lockdown"), is(not(nullValue())));
    }

    @UseReporter(DiffReporter.class)
    @Test
    public void list() throws Exception {
        Approvals.verify(sort(read("list")));
    }

    @UseReporter(DiffReporter.class)
    @Test
    public void tree() throws Exception {
        Approvals.verify(read("tree"));
    }

    private String read(final String goal) throws IOException {
        final URL resource = Resources.getResource(getClass(),
                String.format("%s.%s.actual.txt", getClass().getSimpleName(), goal));
        return Resources.toString(resource, Charsets.UTF_8);
    }

    private static String sort(final String unsorted) {
        final String[] lines = unsorted.split("[\r\n]+");
        Arrays.sort(lines);
        return String.join("\n", lines);
    }

}
