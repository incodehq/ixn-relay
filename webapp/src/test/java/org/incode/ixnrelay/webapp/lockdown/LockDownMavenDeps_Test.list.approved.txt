
   ch.qos.logback:logback-classic:jar:1.2.3:compile
   ch.qos.logback:logback-core:jar:1.2.3:compile
   com.approvaltests:approvaltests-util:jar:2.0.0:test
   com.approvaltests:approvaltests:jar:2.0.0:test
   com.fasterxml.jackson.core:jackson-annotations:jar:2.9.8:compile
   com.fasterxml.jackson.core:jackson-core:jar:2.9.8:compile
   com.fasterxml.jackson.core:jackson-databind:jar:2.9.8:compile
   com.fasterxml.jackson.datatype:jackson-datatype-jdk8:jar:2.9.9:compile
   com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.9.9:compile
   com.fasterxml.jackson.module:jackson-module-parameter-names:jar:2.9.9:compile
   com.fasterxml.woodstox:woodstox-core:jar:5.0.3:compile
   com.fasterxml:classmate:jar:1.4.0:compile
   com.google.guava:guava:jar:19.0:compile
   com.googlecode.json-simple:json-simple:jar:1.1.1:compile
   com.jayway.jsonpath:json-path:jar:2.4.0:test
   com.sun.xml.bind:jaxb-core:jar:2.3.0:compile
   com.sun.xml.bind:jaxb-impl:jar:2.3.0:compile
   com.vaadin.external.google:android-json:jar:0.0.20131108.vaadin1:test
   commons-codec:commons-codec:jar:1.7:compile
   io.micrometer:micrometer-core:jar:1.1.6:compile
   javax.annotation:javax.annotation-api:jar:1.3.2:compile
   javax.jms:javax.jms-api:jar:2.0.1:compile
   javax.validation:validation-api:jar:2.0.1.Final:compile
   javax.ws.rs:javax.ws.rs-api:jar:2.1.1:compile
   joda-time:joda-time:jar:2.10.3:compile
   junit:junit:jar:4.12:test
   net.bytebuddy:byte-buddy-agent:jar:1.9.16:test
   net.bytebuddy:byte-buddy:jar:1.9.16:test
   net.minidev:accessors-smart:jar:1.2:test
   net.minidev:json-smart:jar:2.3:test
   org.apache.activemq:activemq-camel:jar:5.15.10:compile
   org.apache.activemq:activemq-client:jar:5.15.10:compile
   org.apache.activemq:activemq-jms-pool:jar:5.15.10:compile
   org.apache.activemq:activemq-pool:jar:5.15.10:compile
   org.apache.activemq:activemq-spring:jar:5.15.10:compile
   org.apache.camel:camel-core-starter:jar:2.24.2:compile
   org.apache.camel:camel-core:jar:2.24.2:compile
   org.apache.camel:camel-jms:jar:2.24.2:compile
   org.apache.camel:camel-spring-boot-starter:jar:2.24.2:compile
   org.apache.camel:camel-spring-boot:jar:2.24.2:compile
   org.apache.camel:camel-spring:jar:2.24.2:compile
   org.apache.camel:camel-test-spring:jar:2.24.2:test
   org.apache.camel:camel-test:jar:2.24.2:test
   org.apache.commons:commons-pool2:jar:2.6.2:compile
   org.apache.cxf:cxf-core:jar:3.2.10:compile
   org.apache.cxf:cxf-rt-frontend-jaxrs:jar:3.2.10:compile
   org.apache.cxf:cxf-rt-rs-client:jar:3.2.10:compile
   org.apache.cxf:cxf-rt-transports-http-hc:jar:3.2.10:compile
   org.apache.cxf:cxf-rt-transports-http:jar:3.2.10:compile
   org.apache.geronimo.specs:geronimo-j2ee-management_1.1_spec:jar:1.0.1:compile
   org.apache.geronimo.specs:geronimo-jms_1.1_spec:jar:1.1.1:compile
   org.apache.geronimo.specs:geronimo-jta_1.0.1B_spec:jar:1.1.1:compile
   org.apache.httpcomponents:httpasyncclient:jar:4.1.4:compile
   org.apache.httpcomponents:httpclient:jar:4.5.2:compile
   org.apache.httpcomponents:httpcore-nio:jar:4.4.12:compile
   org.apache.httpcomponents:httpcore:jar:4.4.12:compile
   org.apache.isis.core:isis-core-schema:jar:1.17.0:compile
   org.apache.logging.log4j:log4j-api:jar:2.11.2:compile
   org.apache.logging.log4j:log4j-to-slf4j:jar:2.11.2:compile
   org.apache.tomcat.embed:tomcat-embed-core:jar:9.0.24:compile
   org.apache.tomcat.embed:tomcat-embed-el:jar:9.0.24:compile
   org.apache.tomcat.embed:tomcat-embed-websocket:jar:9.0.24:compile
   org.apache.ws.xmlschema:xmlschema-core:jar:2.2.4:compile
   org.apache.xbean:xbean-spring:jar:4.5:compile
   org.assertj:assertj-core:jar:3.6.2:test
   org.codehaus.woodstox:stax2-api:jar:3.1.4:compile
   org.fusesource.hawtbuf:hawtbuf:jar:1.11:compile
   org.hamcrest:hamcrest-core:jar:1.3:compile
   org.hamcrest:hamcrest-library:jar:1.3:compile
   org.hdrhistogram:HdrHistogram:jar:2.1.9:compile
   org.hibernate.validator:hibernate-validator:jar:6.0.17.Final:compile
   org.incode.ixn-relay:ixn-relay-spi-mq:jar:1.0.0-SNAPSHOT:compile
   org.incode.ixn-relay:ixn-relay-spi:jar:1.0.0-SNAPSHOT:compile
   org.isisaddons.module.publishmq:isis-module-publishmq-dom-outboxclient:jar:1.17.0.20191003-1233-47ac5d22:compile
   org.isisaddons.module.publishmq:isis-module-publishmq-outbox-canonical:jar:1.17.0.20191003-1233-47ac5d22:compile
   org.jboss.logging:jboss-logging:jar:3.3.3.Final:compile
   org.jboss.spec.javax.ws.rs:jboss-jaxrs-api_2.0_spec:jar:1.0.0.Final:compile
   org.jolokia:jolokia-core:jar:1.6.2:compile
   org.latencyutils:LatencyUtils:jar:2.0.3:compile
   org.mockito:mockito-core:jar:2.23.4:test
   org.objenesis:objenesis:jar:2.4:test
   org.ow2.asm:asm:jar:5.0.4:test
   org.projectlombok:lombok:jar:1.18.10:provided
   org.skyscreamer:jsonassert:jar:1.5.0:test
   org.slf4j:jcl-over-slf4j:jar:1.7.28:compile
   org.slf4j:jul-to-slf4j:jar:1.7.28:compile
   org.slf4j:slf4j-api:jar:1.7.28:compile
   org.springframework.boot:spring-boot-actuator-autoconfigure:jar:2.1.8.RELEASE:compile
   org.springframework.boot:spring-boot-actuator:jar:2.1.8.RELEASE:compile
   org.springframework.boot:spring-boot-autoconfigure:jar:2.1.8.RELEASE:compile
   org.springframework.boot:spring-boot-configuration-processor:jar:2.1.8.RELEASE:compile
   org.springframework.boot:spring-boot-starter-activemq:jar:2.1.8.RELEASE:compile
   org.springframework.boot:spring-boot-starter-actuator:jar:2.1.8.RELEASE:compile
   org.springframework.boot:spring-boot-starter-json:jar:2.1.8.RELEASE:compile
   org.springframework.boot:spring-boot-starter-logging:jar:2.1.8.RELEASE:compile
   org.springframework.boot:spring-boot-starter-test:jar:2.1.8.RELEASE:test
   org.springframework.boot:spring-boot-starter-tomcat:jar:2.1.8.RELEASE:compile
   org.springframework.boot:spring-boot-starter-web:jar:2.1.8.RELEASE:compile
   org.springframework.boot:spring-boot-starter:jar:2.1.8.RELEASE:compile
   org.springframework.boot:spring-boot-test-autoconfigure:jar:2.1.8.RELEASE:test
   org.springframework.boot:spring-boot-test:jar:2.1.8.RELEASE:test
   org.springframework.boot:spring-boot:jar:2.1.8.RELEASE:compile
   org.springframework:spring-aop:jar:5.1.9.RELEASE:compile
   org.springframework:spring-beans:jar:5.1.9.RELEASE:compile
   org.springframework:spring-context:jar:5.1.9.RELEASE:compile
   org.springframework:spring-core:jar:5.1.9.RELEASE:compile
   org.springframework:spring-expression:jar:5.1.9.RELEASE:compile
   org.springframework:spring-jcl:jar:5.1.9.RELEASE:compile
   org.springframework:spring-jms:jar:5.1.9.RELEASE:compile
   org.springframework:spring-messaging:jar:5.1.9.RELEASE:compile
   org.springframework:spring-test:jar:5.1.9.RELEASE:test
   org.springframework:spring-tx:jar:5.1.9.RELEASE:compile
   org.springframework:spring-web:jar:5.1.9.RELEASE:compile
   org.springframework:spring-webmvc:jar:5.1.9.RELEASE:compile
   org.xmlunit:xmlunit-core:jar:2.6.3:test
   org.yaml:snakeyaml:jar:1.23:runtime
The following files have been resolved: